library/ssl_tls13_server.c

```diff
@@ -174,22 +175,32 @@ static int ssl_tls13_get_psk( mbedtls_ssl_context *ssl,
 MBEDTLS_CHECK_RETURN_CRITICAL
 static int ssl_tls13_offered_psks_check_binder_match( mbedtls_ssl_context *ssl,
                                                       const unsigned char *binder,
-                                                      uint16_t binder_len )
+                                                      size_t binder_len )
 {
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
     int psk_type;
 
-    mbedtls_md_type_t md_alg =
-        binder_len == 32 ? MBEDTLS_MD_SHA256 : MBEDTLS_MD_SHA384 ;
-    psa_algorithm_t psa_md_alg = mbedtls_psa_translate_md( md_alg );
+    mbedtls_md_type_t md_alg;
+    psa_algorithm_t psa_md_alg;
     unsigned char transcript[PSA_HASH_MAX_SIZE];
     size_t transcript_len;
-    const unsigned char *psk;
+    unsigned char *psk;
     size_t psk_len;
     unsigned char server_computed_binder[PSA_HASH_MAX_SIZE];
 
     psk_type = MBEDTLS_SSL_TLS1_3_PSK_EXTERNAL;
-
+    switch( binder_len )
+    {
+    case 32:
+        md_alg = MBEDTLS_MD_SHA256;
+        break;
+    case 48:
+        md_alg = MBEDTLS_MD_SHA384;
+        break;
+    default:
+        return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
```

Comment:
```suggestion
        return( MBEDTLS_SSL_ALERT_MSG_DECRYPT_ERROR );
```
This is not an internal error as the binder length here comes from the peer.